test/integration/test_server.cpp

```diff
@@ -24,10 +24,13 @@ bool NO_FAIL(const grpc::Status &status) {
   return status.ok();
 }
 
-bool IS_FAIL(const grpc::Status &status, const std::string &expected_error) {
-  if (!status.ok() && status.error_message() != expected_error) {
-    fprintf(stderr, "Query failed with unexpected message: %s\n",
-            status.error_message().c_str());
+bool IS_FAIL(const grpc::Status &status,
```

Comment:
How is this test related to the change?

Reply:
I tried to use REQUIRE_FAIL to test the change and realized the behavior in IS_FAIL was not useful.

Reply:
I put the strict equality check back, but made it a native Catch2 check, so it prints the diffs correctly. I did keep the new-to-this-PR behavior of failing if the message is wrong.